crates/network-scheduler/src/scheduler.rs

```diff
@@ -94,17 +96,7 @@ impl Scheduler {
         }
         let assigned_chunks = worker_state.assigned_chunks(&self.known_units);
         if ACTIVE_V2_MIN_WORKER_VER.matches(&version) {
-            let dataset_chunks = assigned_chunks
-                .map(|chunk| (chunk.dataset_url, chunk.chunk_str))
-                .into_grouping_map()
-                .collect::<Vec<_>>()
-                .into_iter()
-                .map(|(dataset_url, chunks)| DatasetChunks {
-                    dataset_url,
-                    chunks,
-                })
-                .collect();
-            WorkerStatus::ActiveV2(WorkerAssignment { dataset_chunks })
+            WorkerStatus::ActiveV2(chunks_to_assignment(assigned_chunks))
         } else {
             WorkerStatus::Active(chunks_to_worker_state(assigned_chunks))
```

Comment:
This can be removed already. No workers <0.3.5 supported.

Reply:
Actually there are still 11 workers with those versions. But yes, it's better to drop them